crawl-ref/source/spl-util.cc

```diff
@@ -1136,13 +1137,25 @@ string casting_uselessness_reason(spell_type spell, bool temp)
         if (you.duration[DUR_CONF] > 0)
             return "you're too confused to cast spells.";
 
-        if (!enough_mp(spell_mana(spell), true, false))
+        if (spell_difficulty(spell) > you.experience_level)
+            return "you aren't experienced enough to cast this spell.";
+
+        if (you.has_mutation(MUT_HP_CASTING))
+        {
+            // TODO: deduplicate with enough_hp()
+            if (you.duration[DUR_DEATHS_DOOR])
+                return "you cannot pay life while functionally dead.";
+            if (!enough_hp(spell_mana(spell), true, false))
+                return "you don't have enough health to cast this spell.";
+        } else if (!enough_mp(spell_mana(spell), true, false))
             return "you don't have enough magic to cast this spell.";
 
         if (spell == SPELL_SUBLIMATION_OF_BLOOD
             && you.magic_points == you.max_magic_points)
         {
-            return "your reserves of magic are already full.";
+            if (you.max_magic_points)
+                return "your reserves of magic are already full.";
+            return "your magic and health are inextricable.";
```

Comment:
It's possible in principle for a non-dj to have 0mp. (Of course, that kind of scenario does make it hard to have sublimation memorised...)

Reply:
Yeah, good catch. I've been using `HP_CASTER` for these sorts of situations elsewhere, so I'll switch it to be consistent.